storage/innobase/srv/srv0srv.cc

```diff
@@ -2668,6 +2671,55 @@ srv_master_do_shutdown_tasks(
 	return(n_bytes_merged || n_tables_to_drop);
 }
 
+void
+srv_enable_temp_encryption_if_set()
+{
+	fil_space_t*	space;
+
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+
+	if (is_system_tablespace(srv_tmp_space.space_id())) {
+		/* there is no separate tablespace for temporary tables */
+		return;
+	}
+
+	space = fil_space_get(srv_tmp_space.space_id());
+	ut_ad(fsp_is_system_temporary(space->id));
+
+	if (srv_tmp_tablespace_encrypt) {
+
+		if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
+			byte	encrypt_info[ENCRYPTION_INFO_SIZE_V2];
+			mtr_t	mtr;
+
+			space->flags |= FSP_FLAGS_MASK_ENCRYPTION;
+
+			memset(encrypt_info, 0, ENCRYPTION_INFO_SIZE_V2);
+
+			mtr_start(&mtr);
+			mtr_set_log_mode(&mtr, MTR_LOG_NO_REDO);
+			mtr.set_named_space(space->id);
+
+			space = mtr_x_lock_space(space->id, &mtr);
+
+			if (!fsp_header_rotate_encryption(space,
+							  encrypt_info,
+							  &mtr)) {
+				mtr_commit(&mtr);
+
+				ib::error() << "Can't set temporary"
```

Comment:
Setting the variable and making the change are far from each other, how to communicate this to user? Who would not necessarily check the error log. Is ib::fatal() an option here?

Reply:
agree, ib::fatal() makes sense to me